### PR TITLE
VACMS-7337: Rename sections and update behat tests.

### PR DESCRIPTION
**Pull request by msbtterswrth**
## Description

Relates to #7337 .

## Testing done
Updated behat/phpunit tests to replace old section names/urls with new ones. Ran tests locally to ensure they passed. 

## Screenshots
Terminal output:
```
2021-12-27T22:46:23.673Z - 61c21083690c6866888e43dd# /bin/sh -c drush deploy
 [notice] Database updates start.
 ----------- ----------- --------------- --------------------------------------- 
  Module      Update ID   Type            Description                            
 ----------- ----------- --------------- --------------------------------------- 
  va_gov_db   8033        hook_update_n   8033 - Rename four key section names.  
 ----------- ----------- --------------- --------------------------------------- 


 >  Do you wish to run the specified pending updates? (yes/no) [yes]:
> >  [notice] Update started: va_gov_db_update_8033
> >  [notice] Vocabulary:administration Term:2 was renamed from 'National Cemetery Administration' to 'NCA'.
> > Vocabulary:administration Term:5 was renamed from 'Veterans Affairs' to 'VACO'.
> > Vocabulary:administration Term:1 was renamed from 'Veterans Benefits Administration' to 'VBA'.
> > Vocabulary:administration Term:3 was renamed from 'Veterans Health Administration' to 'VHA'.
> > 
> >  [notice] Update completed: va_gov_db_update_8033
>  [success] Finished performing updates.
 [success] Cache rebuild start.
```
image

image

image

## QA steps

As a cms engineer, 
1. my automated tests are passing with renamed section titles
As an admin. 
1.  after code released visit https://pr7362-96ya26f2ixuqalvpxyqmsdauwbcjjqvr.ci.cms.va.gov/admin/reports/dblog?type%5B%5D=va_gov_db
    - [x] validate that there are 4 entries showing content changes for terms.
2. Visit  https://pr7362-96ya26f2ixuqalvpxyqmsdauwbcjjqvr.ci.cms.va.gov/admin/structure/taxonomy/manage/administration/overview
   - [x] Validate that you see "NCA"
   - [x] Validate that you see "VACO"
   - [x] Validate that you see "VBA"
   - [x] Validate that you see "VHA"
 3. Visit https://pr7362-96ya26f2ixuqalvpxyqmsdauwbcjjqvr.ci.cms.va.gov/section/veterans-affairs
    - [x] Validate that you are redirected to /section/vaco 

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS team`
- [x] `Sitewide CMS Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Review by swirtSJW — COMMENTED** [review]
Something is strange here.  I don't know how this passed tests because VACO does not exist in prod nor does VHA.

**Comment by msbtterswrth** [comment]
Note: I manually updated the content in the tugboat env to match the expected titles: 

- Veterans Affairs => VACO
- National Cemetery Administration => NCA
- Veterans Benefits Administration => VBA
- Veterans Health Administration => VHA
per @swirtSJW we'll need a hook to automatically update these four terms.

**Review by swirtSJW — DISMISSED** [review]
This is blocked... it did not pass tests naturally... there was a data change that allowed them to pass.  The section title changes should be in a hook_update   so the test changes and the term changes happen in unison.

**Comment by swirtSJW** [comment]
I added the hook_update_N to rename the 4 section terms.  This should get tests passing and carry this change out automatically.

**Comment by swirtSJW** [comment]
@msbtterswrth  Be sure to take a look at the code, and let me know if you want to go through the anatomy of a hook_update_n.   `_va_gov_rename_vocab_term()` is reusable so we can use it the next time we need to change taxonomy term names.

**Comment by swirtSJW** [comment]
This isn't quite ready yet.  I need to add redirect building.

File: docroot/modules/custom/va_gov_db/va_gov_db.install [hunk 1]
```diff
@@ -89,6 +91,80 @@ function _va_gov_db_change_text_field_max_length($field_name, $new_length) {
...
+ *
+ * @param string $old_path
+ *   The original path (source) of the redirect.
+ * @param Drupal\Core\Entity\EntityInterface $entity
+ *   The entity to redirect to.
+ *
+ * @return string
+ *   A message to be returned.
+ */
+function _va_gov_db_create_redirect($old_path, EntityInterface $entity) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_db/va_gov_db.install:150]: This function was not needed for this PR (I originally thought it was), but will be useful soon when drupal becomes the source of redirects.

**Review by JayDarnell — APPROVED** [review]
Code review looks good to me and QA steps pass.

**Comment by JayDarnell** [comment]
@swirtSJW updated based on the base branch and everything looks good.

**Pull request merged**